aws_lambda_powertools/metrics/provider/datadog/datadog.py

```diff
@@ -0,0 +1,291 @@
+            output_list.append(
+                {
+                    "m": metric_name,
+                    "v": single_metric["v"],
+                    "e": single_metric["e"],
+                    "t": single_metric["t"] or list(self.default_tags),
+                },
+            )
+
+        return output_list
+
+    # flush serialized data to output
+    def flush_metrics(self, raise_on_empty_metrics: bool = False) -> None:
+        """Manually flushes the metrics. This is normally not necessary,
+        unless you're running on other runtimes besides Lambda, where the @log_metrics
+        decorator already handles things for you.
+
+        Parameters
+        ----------
+        raise_on_empty_metrics : bool, optional
+            raise exception if no metrics are emitted, by default False
+        """
+        if not raise_on_empty_metrics and len(self.metric_set) == 0:
+            warnings.warn(
+                "No application metrics to publish. The cold-start metric may be published if enabled. "
+                "If application metrics should never be empty, consider using 'raise_on_empty_metrics'",
+                stacklevel=2,
+            )
+
+        else:
+            metrics = self.serialize_metric_set()
+            # submit through datadog extension
+            if lambda_metric and not self.flush_to_log:
+                # use lambda_metric function from datadog package, submit metrics to datadog
+                for metric_item in metrics:  # pragma: no cover
+                    lambda_metric(  # pragma: no cover
+                        metric_name=metric_item["m"],
+                        value=metric_item["v"],
+                        timestamp=metric_item["e"],
+                        tags=metric_item["t"],
+                    )
+            else:
+                # dd module not found: flush to log, this format can be recognized via datadog log forwarder
+                # https://github.com/Datadog/datadog-lambda-python/blob/main/datadog_lambda/metric.py#L77
+                for metric_item in metrics:
+                    print(json.dumps(metric_item, separators=(",", ":")))
+
+        self.clear_metrics()
+
+    def clear_metrics(self):
+        logger.debug("Clearing out existing metric set from memory")
+        self.metric_set.clear()
+
+    def add_cold_start_metric(self, context: LambdaContext) -> None:
+        """Add cold start metric and function_name dimension
+
+        Parameters
+        ----------
+        context : Any
+            Lambda context
+        """
+        logger.debug("Adding cold start metric and function_name tagging")
+        self.add_metric(name="ColdStart", value=1, function_name=context.function_name)
+
+    def log_metrics(
+        self,
+        lambda_handler: Callable[[Dict, Any], Any] | Optional[Callable[[Dict, Any, Optional[Dict]], Any]] = None,
+        capture_cold_start_metric: bool = False,
+        raise_on_empty_metrics: bool = False,
+        **kwargs,
```

Comment:
```suggestion
        default_tags: Dict | None = None,
        **kwargs,
```

it'll hint in IDE giving a better experience.

Reply:
Done